packages/js/src/novu.test.ts

```diff
@@ -1,107 +1,78 @@
+afterAll(() => jest.restoreAllMocks());
 
 describe('Novu', () => {
+  const applicationIdentifier = 'foo';
+  const subscriberId = 'bar';
+
   beforeEach(() => {
-    jest.clearAllMocks();
+    // @ts-ignore
+    global.fetch.mockImplementation(mockFetch) as jest.Mock;
   });
 
-  describe('lazy session initialization', () => {
-    test('should call the queued notifications.list after the session is initialized', async () => {
+  describe('http client', () => {
+    test('should call the notifications.list after the session is initialized', async () => {
       const options = {
         limit: 10,
         offset: 0,
       };
-      const novu = new Novu({ applicationIdentifier: 'applicationIdentifier', subscriberId: 'subscriberId' });
-      const { data } = await novu.notifications.list(options);
 
-      expect(post).toHaveBeenCalledTimes(1);
-      expect(getFullResponse).toHaveBeenCalledWith('/inbox/notifications?limit=10');
-      expect(data).toEqual({
-        notifications: mockNotificationsResponse.data,
-        hasMore: mockNotificationsResponse.hasMore,
-        filter: mockNotificationsResponse.filter,
+      const novu = new Novu({ applicationIdentifier, subscriberId });
+      expect(fetch).toHaveBeenNthCalledWith(1, 'https://api.novu.co/v1/inbox/session/', {
+        method: 'POST',
+        body: JSON.stringify({ applicationIdentifier, subscriberId }),
+        headers: {
+          'Content-Type': 'application/json',
+          'Novu-API-Version': '2024-06-26',
+          'User-Agent': '@novu/js@test',
+        },
       });
-    });
 
-    test('should call the notifications.list right away when session is already initialized', async () => {
-      const options: ListNotificationsArgs = {
-        limit: 10,
-        offset: 0,
-      };
-      const novu = new Novu({ applicationIdentifier: 'applicationIdentifier', subscriberId: 'subscriberId' });
-      // await for session initialization
-      await new Promise((resolve) => {
-        setTimeout(resolve, 10);
+      const { data } = await novu.notifications.list(options);
+      expect(fetch).toHaveBeenNthCalledWith(2, 'https://api.novu.co/v1/inbox/notifications/?limit=10', {
+        method: 'GET',
+        body: undefined,
+        headers: {
+          Authorization: 'Bearer cafebabe',
+          'Content-Type': 'application/json',
+          'Novu-API-Version': '2024-06-26',
+          'User-Agent': '@novu/js@test',
+        },
       });
 
-      const { data } = await novu.notifications.list({ limit: 10, offset: 0 });
-
-      expect(post).toHaveBeenCalledTimes(1);
-      expect(getFullResponse).toHaveBeenCalledWith('/inbox/notifications?limit=10');
       expect(data).toEqual({
         notifications: mockNotificationsResponse.data,
         hasMore: mockNotificationsResponse.hasMore,
         filter: mockNotificationsResponse.filter,
       });
     });
-
-    test('should reject the queued notifications.list if session initialization fails', async () => {
```

Comment:
why we removed these two?

Reply:
I moved these tests to the BaseModule. Did I miss anything?

Reply:
Well, I missed to push the new file 😮‍💨